spec.html

```diff
@@ -9117,17 +9117,20 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
     </emu-clause>
 
     <emu-clause id="sec-createbuiltinfunction" aoid="CreateBuiltinFunction">
-      <h1>CreateBuiltinFunction ( _steps_, _internalSlotsList_ [ , _realm_ [ , _prototype_ ] ] )</h1>
-      <p>The abstract operation CreateBuiltinFunction takes arguments _steps_ and _internalSlotsList_ (a List of names of internal slots) and optional arguments _realm_ and _prototype_. _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
+      <h1>CreateBuiltinFunction ( _steps_, _additionalInternalSlotsList_ [ , _realm_ [ , _prototype_ ] ] )</h1>
+      <p>The abstract operation CreateBuiltinFunction takes arguments _steps_ and optional arguments _additionalInternalSlotsList_ (a List of internal slot names), _realm_, and _prototype_. _additionalInternalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
       <emu-alg>
         1. Assert: _steps_ is either a set of algorithm steps or other definition of a function's behaviour provided in this specification.
         1. If _realm_ is not present, set _realm_ to the current Realm Record.
         1. Assert: _realm_ is a Realm Record.
         1. If _prototype_ is not present, set _prototype_ to _realm_.[[Intrinsics]].[[%Function.prototype%]].
-        1. Let _func_ be a new built-in function object that when called performs the action described by _steps_. The new function object has internal slots whose names are the elements of _internalSlotsList_, and an [[InitialName]] internal slot.
-        1. Set _func_.[[Realm]] to _realm_.
-        1. Set _func_.[[Prototype]] to _prototype_.
+        1. Let _internalSlotsList_ be the internal slots specified in <emu-xref href="#sec-built-in-function-objects"></emu-xref>.
+        1. Append [[InitialName]] to _internalSlotsList_.
+        1. Append each element of _additionalInternalSlotsList_ to _internalSlotsList_.
+        1. Let _func_ be a new built-in function object that when called performs the action described by _steps_. The new function object has internal slots whose names are the elements of _internalSlotsList_.
```

Comment:
As @anba pointed out: if `_func_` is an ECMAScript (i.e. ordinary) function object, it must have all the internal slots listed [here](https://tc39.es/ecma262/#table-internal-slots-of-ecmascript-function-objects), but this operation doesn't mention most of them or say how they're initialized. Of course, that's been true since `CreateBuiltinFunction` was introduced, but this PR appears to be trying for completeness. I think it's somewhat misleading in its current state.

Reply:
I’m not sure how I should specify that given @bakkot’s review in <https://github.com/tc39/ecma262/pull/2115#discussion_r467621952>.

Reply:
If @bakkot continues to object, then I think the solution is for `CreateBuiltinFunction` to not be as specific about slots as you'd like. E.g., you could ditch the construction of `_internalSlotsList_` and then say something like: `The new function object has the internal slots required by either (9.2) or (9.3), as appropriate, and also has internal slots whose names are the elements of _additionalInternalSlotsList_.`

(It *should* be possible to replace "either (9.2) or (9.3), as appropriate" with just "(9.3)", since 9.3 deals with all built-in functions, and covers the ordinary/exotic implementation choice. But I think, as 9.3 currently stands, that would be a little misleading, as readers would only look at the slots explicitly mentioned in 9.3 and not consider those included by reference to 9.2.)